flow-server/src/main/resources/plugins/theme-live-reload-plugin/theme-live-reload-plugin.js

```diff
@@ -44,28 +52,60 @@ class ThemeLiveReloadPlugin {
-              // re-generates theme-[my-theme].generated.js),
-              // otherwise it will get into infinite re-compilation loop.
-              // There might be several theme generated files in the
-              // generated folder, so the condition does not contain the exact
-              // theme name
-              if (file.match(/theme-[\s\S]*?\.generated\.js$/)) {
-                themeGeneratedFileChanged = true;
+          changedFilesPaths.forEach(changedFilePath => {
+            const file = `${changedFilePath}`;
+            // Webpack watches to the changes in theme-[my-theme].generated.js
+            // because it is referenced from theme.js. Changes in this file
+            // should not trigger the theme handling callback (which
+            // re-generates theme-[my-theme].generated.js),
+            // otherwise it will get into infinite re-compilation loop.
+            // There might be several theme generated files in the
+            // generated folder, so the condition does not contain the exact
+            // theme name
+            const themeGeneratedFileChangedNow = file.match(this.themeGeneratedFileRegexp);
+            if (!themeGeneratedFileChanged && themeGeneratedFileChangedNow) {
+              themeGeneratedFileChanged = true;
+            }
+
+            const timestamp = changedFilesMap[changedFilePath];
+            // null or negative timestamp means file delete
+            if (timestamp === null || timestamp < 0) {
+              if (themeGeneratedFileChangedNow) {
+                themeGeneratedFileDeleted = true;
+              } else if (file.match(this.componentStyleFileRegexp)) {
+                deletedComponentStyleFile = file;
               }
-            });
-          if (!themeGeneratedFileChanged) {
+            }
+          });
+          // This is considered as a workaround for
+          // https://github.com/vaadin/flow/issues/9948: delete component
+          // styles and theme generated file in one run to not have webpack
+          // compile error
+          if (deletedComponentStyleFile && !themeGeneratedFileDeleted) {
+            logger.warn("Custom theme component style sheet '" + deletedComponentStyleFile + "' has been deleted.\n" +
+              "Be sure you've also deleted 'generated/theme-" + this.themeName + ".generated.js' in one run (simultaneously) with the component stylesheet'.\n" +
+              "Otherwise, it would cause webpack 'no such file or directory' compilation error, because component style sheets are referenced in " +
+              "'generated/theme-" + this.themeName + ".generated.js'.\n" +
+              "If you encounter a 'no such file or directory' error shown on the overlay in your application, " +
+              "click on the overlay (or refresh the browser page), and it should disappear.\n" +
+              "You can then continue working on your application and theming.\nIf it doesn't help, you need to restart the application.");
```

Comment:
```suggestion
            logger.warn("Custom theme component style sheet '" + deletedComponentStyleFile + "' has been deleted.\n\n" +
              "You should also delete './frontend/generated/theme-" + this.themeName + ".generated.js' (simultaneously) with the component stylesheet'.\n" +
              "Otherwise it will cause a webpack compilation error 'no such file or directory', as component style sheets are referenced from " +
              "'./frontend/generated/theme-" + this.themeName + ".generated.js'.\n\n" +
              "If you encounter a 'no such file or directory' error in your application, " +
              "just click on the overlay (or refresh the browser page), and it should disappear.\n\n" +
              "It should then be possible to continue working on the application and theming.\n" +
              "If it doesn't help, you need to restart the application.");
```

Here I'm also questioning the use of themeName as what if the component style deleted was from a parent theme? 
This should lead to the same exception as deleting one from the current theme as the parent theme is referenced from the current theme.

Reply:
Detecting changes in parent theme folder is a bit problematic, since the only current theme component folder is being watched. Thus, even though the live reload gets triggered due to changes in parent theme files, the changed files wouldn't be passed to the `compiler.watchFileSystem.watcher.mtimes`, and the application wouldn't know that the parent component style sheet has been actually deleted.

The solution is to make all theme component folders to be watched by watch plugin by passing there a list of theme component paths regexps,  not an exact theme path. It will solve the issue with live reload after theme swapping, by the way. But of course, it will trigger a live reload upon non-relevant changes in other themes folders. 

WDYT @caalador ?

Or I could somehow extract the parent theme name from theme.json in webpack.generated.js. But I don't know what exactly the current theme directory is at that point.

Reply:
I would be possible at line:176
```
const flowFrontendThemesFolder = path.resolve(flowFrontendFolder, 'themes');
let themeName = undefined;
if (devMode) {
  // Current theme name is being extracted from theme.js located in frontend
  // generated folder
  themeName = extractThemeName(frontendGeneratedFolder);
}
```
Where we know the themes folder and the themeName to get the theme.json for parsing, but perhaps that change would be for it's own PR.

Reply:
`flowFrontendThemesFolder`  refers to `target/flow-frontend/themes/` as I know. And I'm not sure we should watch it, as it comes from themes packed in JAR.

I believe a proper way is to look through the `themeProjectFolders` and find an appropriate parent theme there.
Can be refactored so as to use the existing lookup code in `theme-handle.js`.

But for this PR, as you suggest, I am only fixing the rest of the issues.